src/monitoring/monitored-certificate.ts

```diff
@@ -0,0 +1,113 @@
+import { ICertificate } from '@aws-cdk/aws-certificatemanager';
+import { ComparisonOperator, Metric, MetricOptions, Statistic, TreatMissingData } from '@aws-cdk/aws-cloudwatch';
+import { Construct, Duration, Stack } from '@aws-cdk/core';
+import type { IMonitoredCertificate, Monitoring } from '.';
+import { CertificateMonitor } from './certificate-monitor';
+
+export interface MonitoredCertificateProps {
+  /**
+   * The ACM certificate to be monitored.
+   */
+  readonly certificate: ICertificate;
+
+  /**
+   * The DNS name of an endpoint using the monitored certificate.
+   */
+  readonly domainName: string;
+
+  /**
+   * The monitoring instance that owns this certificate monitor.
+   */
+  readonly monitoring: Monitoring;
+
+  /**
+   * The namespace within which to emit the metric.
+   *
+   * @default Stack.of(this).stackName
+   */
+   readonly metricNamespace?: string;
+
+   /**
+    * The name of the CloudWatch metric to be emitted.
+    *
+    * @default 'DaysToExpiry'
+    */
+   readonly metricName?: string;
+}
+
+export class MonitoredCertificate extends Construct implements IMonitoredCertificate {
+  private readonly customMetricName: string;
+  private readonly customMetricNamespace: string;
+
+  public constructor(scope: Construct, id: string, private readonly props: MonitoredCertificateProps) {
+    super(scope, id);
+
+    this.customMetricName = props.metricName ?? 'DaysToExpiry';
+    this.customMetricNamespace = props.metricNamespace ?? Stack.of(this).stackName;
+
+    const dynamicMonitor = new CertificateMonitor(this, `${props.certificate.node.addr}::Monitor`, {
```

Comment:
Redundant. Scope name is intrinsically part of the ID:

```suggestion
    const dynamicMonitor = new CertificateMonitor(this, `Monitor`, {
```

Reply:
Oh yeah - traces of a refactor :P